src/query/storages/fuse/src/operations/gc.rs

```diff
@@ -393,10 +409,28 @@ impl FuseTable {
     ) -> Result<()> {
         let fuse_file = Files::create(ctx.clone(), self.operator.clone());
         let locations = Vec::from_iter(locations_to_be_purged);
-        self.clean_cache(&locations);
+        // self.clean_cache(&locations);
         fuse_file.remove_file_in_batch(&locations).await
     }
 
+    // Purge file by location chunks.
+    async fn try_purge_location_files_and_cache<T>(
+        &self,
+        ctx: Arc<dyn TableContext>,
+        locations_to_be_purged: HashSet<String>,
+    ) -> Result<()>
+    where
+        T: CachedMeta<T>,
+    {
+        if let Some(cache) = T::cache() {
+            for loc in locations_to_be_purged.iter() {
+                cache.evict(loc);
+            }
```

Comment:
Can you move the eviction to `try_purge_location_files`?

Reply:
Else all LGTM.

Reply:
looks like we should keep the cache eviction logic in the method
    `async fn try_purge_location_files_and_cache<T: CachedMeta<T>>(...) -> Result<()>`
    
`try_purge_location_files` is for 
- remove files only
- called by removals that (currently) caches are not involved
   e.g. removing the data blocks